include/api.php

```diff
@@ -114,12 +114,12 @@ function pll_default_language( $field = 'slug' ) {
  *
  * @param int                 $post_id Post ID.
  * @param PLL_Language|string $lang    Optional language (object or slug), defaults to the current language.
- * @return int|false The translation post ID if exists, otherwise the passed ID. False if the passed object has no language or if the language doesn't exist.
+ * @return int|false The translation post ID if exists, 0 if not translated or if the post has no language. False if the language doesn't exist.
```

Comment:
In my opinion, we should stop returning `false` and return `0` instead. There is no point changing the type except messing the doc. In fact here the doc is still incorrect.
[The method returns 0 when the language doesn't exist](https://github.com/polylang/polylang/blob/3.6.1/include/translated-object.php#L334-L337) and false [only when no translation's found](https://github.com/polylang/polylang/blob/3.6.1/include/translated-object.php#L312).

Reply:
I also added tests for unexisting language in 0e58e701.